### ISSUE TITLE
`chdir': No such file or directory @ dir_chdir - /home/appmboprod/public_html (Errno::ENOENT)

### ISSUE THREAD

**Issue by sadiqmmm** [puma/puma#904]
![screenshot from 2016-02-26 16 23 40](https://cloud.githubusercontent.com/assets/672989/13350366/94255294-dca5-11e5-96bc-013cfa6857a0.png)

Puma Version 3.0.1 
Rails Version  4.2.5.1
Ruby Version 2.3.0
Ubuntu 14.04

**Comment by evanphx** [comment]
It's only following what you asked it to do, so you must have specified that directory but it's not there?

**Comment by irobayna** [comment]
After upgrading gem to `3.0.2` I get the same error.  

No configuration changes from previous versions 2.x.x -- something has changed which obviusly is taking some settings in `config\puma.rb` in `development` mode which are only intended for `production` (at least on my case)

**Comment by evanphx** [comment]
Can you post our config?

**Comment by irobayna** [comment]
```
#!/usr/bin/env puma
rackup '/home/deployer/apps/xyzAPI/current/config.ru'
rails_env = ENV['RAILS_ENV'] || 'production'
basedir = '/home/deployer/apps/xyzAPI'
directory "#{basedir}/current"
environment 'production'
daemonize true

bind  "unix://#{basedir}/shared/tmp/puma/puma.sock"
pidfile "#{basedir}/shared/tmp/pids/puma.pid"
state_path "#{basedir}/shared/tmp/pids/puma.state"

threads 8,32
workers 3
```

**Comment by evanphx** [comment]
@irobayna What output do you get? Where is puma trying to change the directory to? Is it not that `basedir/current`?

**Comment by sadiqmmm** [comment]
#!/usr/bin/env puma

#start puma with:
#RAILS_ENV=production bundle exec puma -C ./config/puma.rb

application_path = '/home/appmboprod/public_html'
railsenv = 'production'
directory application_path
environment railsenv
daemonize true
#pidfile "#{application_path}/tmp/pids/puma-#{railsenv}.pid"
#state_path "#{application_path}/tmp/pids/puma-#{railsenv}.state"
#stdout_redirect
#{}"#{application_path}/log/puma-#{railsenv}.stdout.log",
#{}"#{application_path}/log/puma-#{railsenv}.stderr.log"
threads 0, 16
workers 2
preload_app!
#bind "unix://#{application_path}/tmp/sockets/#{railsenv}.socket"
bind 'tcp://0.0.0.0:3009'

on_worker_boot do
  ActiveSupport.on_load(:active_record) do
    ActiveRecord::Base.establish_connection
  end
end

**Comment by sadiqmmm** [comment]
From the above puma.rb file script.

Removing the

>  "directory application_path"

puma is working correctly.

**Comment by irobayna** [comment]
@evanphx the problem is when I run on ```development``` mode

Something has changed from ```2.16.0``` to latest release (have tried ```3.0.2``` and ```3.1.0```)

Here is my output after reluctantly disabling the ```directory``` variable in the config file:

```
09:52:14 rails.1   | => Booting Puma
09:52:14 rails.1   | => Rails 4.2.5.2 application starting in development on http://localhost:3000
09:52:14 rails.1   | => Run `rails server -h` for more startup options
09:52:14 rails.1   | => Ctrl-C to shutdown server
09:52:15 rails.1   | [80022] Puma starting in cluster mode...
09:52:15 rails.1   | [80022] * Version 3.1.0 (ruby 2.3.0-p0), codename: El Niño Winter Wonderland
09:52:15 rails.1   | [80022] * Min threads: 8, max threads: 24
09:52:15 rails.1   | [80022] * Environment: development
09:52:15 rails.1   | [80022] * Process workers: 3
09:52:15 rails.1   | [80022] * Preloading application
09:52:15 rails.1   | [80022] * Listening on tcp://localhost:3000
09:52:15 rails.1   | [80022] * Daemonizing...
09:52:15 rails.1   | Exiting
09:52:15 rails.1   | Exiting
09:52:15 rails.1   | exited with code 0
```

**Comment by irobayna** [comment]
@evanphx After spending some time looking at this, I found my configuration works just find on staging and production.  I have added a guard clause to exit if running on ```development```

If my guard clause makes sense, you might want to consider updating the documentation to help other developers who might be using a puma config file and want to use the latest versions of puma.  

BTW, I am running ```3.1.0``` now (development/staging/production) and thank you for all your hard work!
